index.js

```diff
@@ -280,21 +288,26 @@ Client.prototype.getPaymentStatus = function(statusUrl, callback){
       } else {
         callback(response.body);
       }
-
   });
 };
 
 Client.prototype._getAndHandlePaymentStatus = function(statusUrl, callback, loopFunction){
   var self = this;
-  self.getPaymentStatus(statusUrl, function(error, response){
-    if(error){
-      callback(error, null);
-      return loopFunction(statusUrl, callback, loopFunction);
+  self.getPaymentStatus(statusUrl, function(error, response) {
+    if (error) {
+      if (error.response && error.response.body && error.response.body.error === 'txnNotFound') {
+        setTimeout(function() {
+          loopFunction(statusUrl, callback, loopFunction);
+        }, 100);
```

Comment:
This looks like an arbitrary number. I would do couple things here:
- increase the timeout, retrying every 100ms is too often and probably won't be that useful
- ideally we'd have an exponential back-off, but for now increasing the timeout to a second is fine
- move the timeout to a constant in the top of the file and reference it here